native/src/components/TtsContainer.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { createContext, ReactElement, useCallback, useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { AppState } from 'react-native'
+import Tts from 'react-native-tts'
+
+import buildConfig from '../constants/buildConfig'
+import { AppContext } from '../contexts/AppContextProvider'
+import { reportError } from '../utils/sentry'
+import TtsPlayer from './TtsPlayer'
+
+const unsupportedLanguagesForTts = ['fa', 'ka', 'kmr']
+
+export const isTtsActive = (content: string[] | null, languageCode: string): boolean =>
+  Array.isArray(content) &&
+  content.length > 0 &&
+  Boolean(buildConfig().featureFlags.tts) &&
+  !unsupportedLanguagesForTts.includes(languageCode)
+
+export type TtsContextType = {
+  visible: boolean
+  setVisible: (visible: boolean) => void
+  sentences: string[] | null
+  setSentences: (sentences: string[]) => void
+}
+
+export const ttsContext = createContext<TtsContextType>({
+  visible: false,
+  setVisible: () => undefined,
+  sentences: [],
+  setSentences: () => undefined,
+})
+
+type TtsContainerProps = {
+  children: ReactElement
+}
+
+const TtsContainer = ({ children }: TtsContainerProps): ReactElement => {
+  const { languageCode } = React.useContext(AppContext)
+  const { t } = useTranslation('layout')
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isExpanded, setIsExpanded] = useState(false)
+  const [sentenceIndex, setSentenceIndex] = useState(0)
+  const [visible, setVisible] = useState(false)
+  const [sentences, setSentences] = useState<string[]>([])
+
+  const unsupportedLanguage = unsupportedLanguagesForTts.includes(languageCode)
+  const maxTitle = 20
+  const title = sentences[0] || t('readAloud')
```

Comment:
This only will show up if the player is showing but the content is gone if you went back to the categories list for example. what do you think @hauf-toni ?

Reply:
My recommendation for the display of the tts-Feature if there's no readable text would be to show a message like "Nothing to read" in place of the function when the button is pressed. We should also disabled the button in this state to avoid unnecessary user interaction.

Is this what you were hoping to find out?

Reply:
Yes Thank you! I didn't expect a fast replay 🙂